xml/tuning-os-tuning.xml

```diff
@@ -86,244 +91,254 @@
+    appendix.
+   </para>
+  </sect2>
+
+  <sect2>
+   <title>I/O tuning - Multiqueue Block I/O</title>
+   <para>
+    The first aspect to tune is to ensure that I/O if flowing in the most
+    optimal pattern. For the test cluster used in this test, that means enable
+    multi-queue block I/O. This is done through adding a boot-time kernel
+    parameter as found in the Section 12.4
+    <link xlink:href="https://documentation.suse.com/sles/15-SP1/html/SLES-all/cha-tuning-io.html#cha-tuning-io-barrier"/>.
+    This is not an action that should be taken unilaterally on clusters that
+    contain spinning media devices due to potential performance degradation for
+    those devices. The general result is that there are multiple I/O queues
+    assigned to the device, allowing more jobs to be handled simultaneously by
+    those such as NVMe and SSD that can service large numbers of requests.
+   </para>
+  </sect2>
+
+  <sect2>
+   <title>SSD Tuning</title>
+   <para>
+    To get the best read performance, it may be necessary to adjust the
+    <option>read_ahead</option> and write cache settings for the SSD devices.
+    In our particular testing environment, disabling write cache and forcing
+    <option>read_ahead</option> to 2&nbsp;MB resulted in the best overall
+    performance.
+   </para>
+   <para>
+    Before tuning, it is important to check the default values and measure the
+    differences in performance compared to the baseline.
+   </para>
+   <para>
+    By placing the following file in <filename>/etc/udev/rules.d</filename> the
+    device is detected by the model name shown in
+    <filename>/sys/block/{devname}/device/model</filename> and assigned to
+    disable write caching and setting the <option>read_ahead_kb</option> to
+    2&nbsp;MB.
+   </para>
 <screen>
   /etc/udev/rules.d/99-ssd.rules
 
   # Setting specific kernel parameters for a subset of block devices (Intel SSDs)
   SUBSYSTEM=="block", ATTRS{model}=="INTEL SSDS*", ACTION=="add|change", ATTR{queue/read_ahead_kb}="2048"
   SUBSYSTEM=="block", ATTRS{model}=="INTEL SSDS*", ACTION=="add|change", RUN+="/sbin/hdparm -W 0 /dev/%k"
 </screen>
-   </sect2>
-   <sect2>
-     <title>Network Stack and Device Tuning</title>
-     <para>
-       Proper tuning of the network stack can substantially add to improving the
-       latency and throughput of the cluster. A full script for the testing
-       we performed can be found in <xref linkend="tuning-appendix-c"/>.
-     </para>
+  </sect2>
+
+  <sect2>
+   <title>Network Stack and Device Tuning</title>
+   <para>
+    Proper tuning of the network stack can substantially add to improving the
+    latency and throughput of the cluster. A full script for the testing we
+    performed can be found in <xref linkend="tuning-appendix-c"/>.
+   </para>
+   <para>
+    The first and most impactful tuning is to utilize jumbo frame packets. For
+    this to be done, all interfaces utilizing the cluster must be set to the
```

Comment:
This sounds strange to me.

Would "all interfaces used in the cluster" be better?

If it does not refer to internal comms between cluster nodes, then:

"all interfaces used to access the cluster"... but this seems less likely to me.